contentcuration/contentcuration/utils/publish.py

```diff
@@ -403,7 +403,13 @@ def process_assessment_metadata(ccnode, kolibrinode):
     randomize = exercise_data.get('randomize') if exercise_data.get('randomize') is not None else True
     assessment_item_ids = [a.assessment_id for a in assessment_items]
 
-    mastery_model = {'type': exercise_data.get('mastery_model') or exercises.M_OF_N}
+    exercise_data_type = ""
+    if exercise_data.get('mastery_model'):
+        exercise_data_type = exercise_data.get('mastery_model')
+    if exercise_data.get('option') or exercise_data.get('option').get('completion_criteria') or exercise_data.get('option').get('completion_criteria').get('mastery_model'):
```

Comment:
These should be `and` not `or` here, otherwise you are trying to access `.get` from `None`, if `exercise_data.get('option')` is `None`.

I think if you fix this, the tests should pass, and we can merge this PR. We'll manage the rest in follow up work.

Reply:
This is changed